### PR TITLE
Update GHC to 7.8.4

### PR DESCRIPTION
**Pull request by pthariensflame**
See #35476 for context.

### CODE REVIEW

File: Library/Formula/ghc.rb [hunk 1]
```diff
@@ -22,30 +21,26 @@ class Ghc < Formula
 
   if build.build_32_bit? || !MacOS.prefer_64_bit?
     resource "binary" do
-      url "https://www.haskell.org/ghc/dist/7.4.2/ghc-7.4.2-i386-apple-darwin.tar.bz2"
-      sha256 "80c946e6d66e46ca5d40755f3fbe3100e24c0f8036b850fd8767c4f9efd02bef"
+      url "https://downloads.haskell.org/~ghc/7.6.3/ghc-7.6.3-i386-apple-darwin.tar.bz2"
+      sha256 "cc298a03b5450e0cc93cf6929a47a31165733348d2d98f50627ea69532c19391"
```
Comment (mikemcquaid) [inline Library/Formula/ghc.rb]: Don't bump these; we want to keep on an old version to support old versions of OSX.

Comment (pthariensflame) [inline Library/Formula/ghc.rb]: Alright, I'll leave it as 7.4.2 and just update the download path, then.

**Comment by pthariensflame** [comment]
Thanks!

**Comment by jacknagel** [comment]
Pulled.

I dropped the head URL; I can't find the thread right now, but there was some earlier discussion and some people involved with ghc preferred we not package it.

**Comment by pthariensflame** [comment]
Understood.  Thanks!

**Pull request closed without merging**